fiona/__init__.py

```diff
@@ -67,21 +67,62 @@
 import os
 import sys
 import warnings
-
+import platform
 from six import string_types
 from collections import OrderedDict
-
-try:
-    from pathlib import Path
-except ImportError:  # pragma: no cover
-    class Path:
-        pass
+from pathlib import Path
 
 # TODO: remove this? Or at least move it, flake8 complains.
 if sys.platform == "win32":
     libdir = os.path.join(os.path.dirname(__file__), ".libs")
     os.environ["PATH"] = os.environ["PATH"] + ";" + libdir
 
+try:
+
+    from fiona.ogrext import _bounds, _listlayers, FIELD_TYPES_MAP, _remove, \
+        _remove_layer
```

Comment:
If this line is duplicated we need to remember to change both lines if we want to amend the imports in the future. Instead you could import `fiona.ogrext` only as a test, then have the real imports just one after the import succeeds:
```python
try:
    import fiona.ogrext
except ImportError as e:
    if ...:
        import fiona.ogrext
    else:

from fiona.ogr import _bounds, _listlayers, ...
```